registry/storage/driver/swift/swift.go

```diff
@@ -0,0 +1,533 @@
+// Package swift provides a storagedriver.StorageDriver implementation to
+// store blobs in Openstack Swift object storage.
+//
+// This package leverages the ncw/swift client library for interfacing with
+// Swift.
+//
+// Because Swift is a key, value store the Stat call does not support last modification
+// time for directories (directories are an abstraction for key, value stores)
+package swift
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	gopath "path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/ncw/swift"
+
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/base"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+)
+
+const driverName = "swift"
+
+const defaultChunkSize = 5 * 1024 * 1024
```

Comment:
In my experience with the s3 driver, 5 mb might be too small of a default chunk size. You should test to see if making it slightly bigger improves the speed of a large upload. The minimum chunk size for which you get optimal bandwidth utilization will of course be dependent on MANY circumstances, but I suspect that 5mb will be too small for most conventional situations.